HDF5 docs and doc instructions (#102)
mggg/GerryChainJulia: docs/instructions.md

## How to add documentation to the GerryChain Julia project

### Adding a Doc-string
We use `Documenter.jl` to generate our documentation. For `Documenter` to recognize
a doc-string, you should add the doc-string _before_ the function. Lets say you
have a function that adds two numbers. You would want to add documentation in the
following way:

````
"""
    MyAdd(arg₁::Int,
          arg₂::Int)::Int

Adds the two argument integers. *Returns* the result as an Int.
"""
MyAdd(arg₁::Int,
      arg₂::Int)::Int
    arg₁ + arg₂
end
````

> The doc-string inside the tripe quotation marks will render as MarkDown, so you

Reviewer: ```suggestion
The doc-string inside the triple quotation marks will render as MarkDown, so you
```

can also use other MarkDown styles like _italics_ or **bold** there if you want.

### Rendering the doc-string in the documentation page
Each page in the documentation is a different `.md` file in `docs/src/`. Say you
want to add documentation of `your_function()` to `doc_file.md`. Then, you would
go to `docs/src/doc_file.md` and add the documentation where you would want it
to appear like this:

````
```@docs
save_scores_to_json
```
````

That's it! You might want to add something more about this function, which you
could add simply as MarkDown text underneath it. Please check out any of the
`.md` files in `docs/src` for examples.

### Rendering the documentation locally
You will probably want to see how your added documentation looks like before you
make a Pull Request. To do that:  
1. Go to the root of this repository locally in your terminal.
2. Enter `julia --project=. docs/make.jl` . This will generate the HTML files
   locally in the `docs/build` directory.
3. Enter `python -m http.server --bind localhost` . This will run a local server
   that hosts the documentation.
4. The terminal should say that the documentation is being served at some port number.

>    Use your favorite browser to go to that port (for eg, it looks like `localhost:8000`.

Reviewer: ```suggestion
   Use your favorite browser to go to that port (e.g., `localhost:8000`.
```
